source/yats/include/yats/task_configurator.h

```diff
@@ -114,6 +131,18 @@ class task_configurator : public abstract_task_configurator
         return nullptr;
     }
 
+    template <typename LocalTask = Task>
+    static std::enable_if_t<has_thread_constraints_v<LocalTask>, thread_group> default_thread_constraints()
+    {
+        return LocalTask::thread_constraints();
+    }
+
+    template <typename LocalTask = Task>
+    static std::enable_if_t<!has_thread_constraints_v<LocalTask>, thread_group> default_thread_constraints()
+    {
+        return thread_group::any_thread();
```

Comment:
add a default constructor to thread group. If we do not specify this exact thread constraint here and use the default constructor the task_configurator does not need to know any specific constraint.